app/main/viewmanager.ts

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+import { BrowserWindow, ipcMain } from 'electron';
+import { View, ViewProps } from './view';
+
+import ConfigUtil = require('../renderer/js/utils/config-util');
+
+class ViewManager {
+	views: { [key: number]: View };
+	selectedIndex: number;
+
+	constructor() {
+		this.views = {};
+		this.selectedIndex = 0;
+		this.registerIpcs();
+	}
+
+	registerIpcs(): void {
+		ipcMain.on('view-create', (e: Event, props: ViewProps) => {
+			this.create(props);
+		});
+
+		ipcMain.on('view-select', (e: Event, index: number) => {
+			this.select(index);
+		});
+
+		ipcMain.on('view-destroy', (e: Event, index: number) => {
+			this.destroy(index);
+		});
+
+		ipcMain.on('view-destroy-all', () => {
+			this.destroyAll();
+		});
+
+		ipcMain.on('forward-message-view', (e: Event, name: string, ...params: any[]) => {
+			this.views[this.selectedIndex].webContents.send(name, ...params);
+		});
+
+		ipcMain.on('forward-message-all', (e: Event, name: string, ...params: any[]) => {
+			this.forwardMessageAll(name, ...params);
+		});
+
+		ipcMain.on('call-view-function', (e: Event, name: string, ...params: any[]) => {
+			const fn = this.views[this.selectedIndex][name as keyof View];
+			if (typeof fn === 'function') {
+				// Type checking requires spread elements to match up with a rest parameter.
+				// So, using a workaround here.
+				(fn as any)(...params);
+			}
+		});
+	}
+
+	create(props: ViewProps): void {
+		if (this.views[props.index]) {
+			return;
+		}
+		const view = new View(props);
+		this.views[props.index] = view;
+		view.webContents.loadURL(props.url);
+	}
+
+	select(index: number): void {
+		const mainWindow = BrowserWindow.getAllWindows()[0];
+		const view = this.views[index];
+		if (!view || view.isDestroyed()) {
+			console.log('Attempt to select a view that does not exist.');
+			return;
+		}
+		this.selectedIndex = index;
+		if (!view.webContents.getURL()) {
+			const { url } = view;
+			view.webContents.loadURL(url);
+		}
+		mainWindow.setBrowserView(view);
+		view.webContents.focus();
+		this.fixBounds();
+	}
+
+	fixBounds(): void {
+		const SIDEBAR_WIDTH = 54;
```

Comment:
I'd add a comment here mentioning that any updates here should be reflected to css and other way around.